src/polyswarm_api/types/base.py

```diff
@@ -0,0 +1,63 @@
+from jsonschema import validate, ValidationError
+from enum import Enum
+
+from ..log import logger
+from .. import exceptions
+
+
+class BasePSType(object):
+    def __init__(self, polyswarm=None):
+        self.polyswarm = polyswarm
+
+
+class BasePSJSONType(BasePSType):
+    SCHEMA = {
+        'type': ['object', 'array']
+    }
+
+    def __init__(self, json=None, polyswarm=None):
+        super(BasePSJSONType, self).__init__(polyswarm)
+        # this is expensive on thousands of objects
+        # avoid if disabled
+        if polyswarm and polyswarm.validate:
+            self.validate(json)
```

Comment:
I think we should do validation for what we send to the API, not for what we receive. The API is bound to change, and we should have a client that is resilient to these changes. This will break if we make any upgrade to the API. If we have to break stuff, we should break on server-side, not client-side. It would be strange if a API call that returned 200 and worked on server-side failed on client side with `invalid schema`. Also, when people are using, it would not give us pointers on where that schema change might actually have an impact, since it is not going through and generating a real error.

Reply:
This is only done if a `--validate` flag is provided to the client. I've found it quite helpful for testing various things, but we can strip it out if you see no value in it.

Reply:
Nice, a flag is more than enough, I don't think we have to remove them. Being useful for debugging is a gigantic reason for things to exist in my opinion. :+1: